### ISSUE TITLE
F402 is triggered by `__future__.annotations`

### ISSUE THREAD

**Issue by mattmess1221** [astral-sh/ruff#15747]
### Description

F402 is being triggered by `__future__` imports. These imports can probably be safely ignored.

Searched terms: F402, annotations, future

### Reproduction

```py
from __future__ import annotations

for annotations in [["anno1", "anno2"]]:
    print(annotations)
```

Output:

```console
% ruff --version
ruff 0.9.3
% ruff check --select F
f.py:3:5: F402 Import `annotations` from line 1 shadowed by loop variable
  |
1 | from __future__ import annotations
2 |
3 | for annotations in [["anno1", "anno2"]]:
  |     ^^^^^^^^^^^ F402
4 |     print(annotations)
  |

Found 1 error.
```

**Comment by AlexWaygood** [comment]
Yes, putting `from __future__ import annotations` in a module does actually insert an `annotation` symbol into the module's namespace:

```pycon
>>> from __future__ import annotations
>>> annotations
_Feature((3, 7, 0, 'beta', 1), None, 16777216)
```

This is documented at <https://docs.python.org/3/library/__future__.html>.

I appreciate that most people don't ever use the symbol that's inserted as a result of these `import` statements -- the main use case is the "side effect" the import statement has that causes the module to be parsed in a different way. However, your code still might be more readable if you used a different variable name to differentiate your loop variable from the symbol inserted by the import statement.
